From 0e3a8ebab7e367d409ca611af8fee4af26d2205f Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sat, 20 Aug 2022 20:32:07 +0200
Subject: [PATCH 387/389] ASoC: codec: rt5640: Fix hpout restore when lout is
 enabled

When line out and heaphones are enabled, rt5640_lout_event will cause
headpone state restore to fail, by calling hp_amp_power_on again after
rt5640_pmu_depop.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 sound/soc/codecs/rt5640.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sound/soc/codecs/rt5640.c b/sound/soc/codecs/rt5640.c
index 03b9bc47cb61..7a95fe181081 100644
--- a/sound/soc/codecs/rt5640.c
+++ b/sound/soc/codecs/rt5640.c
@@ -997,7 +997,8 @@ static int rt5640_lout_event(struct snd_soc_dapm_widget *w,
 
 	switch (event) {
 	case SND_SOC_DAPM_POST_PMU:
-		hp_amp_power_on(component);
+		//hp_amp_power_on(component);
+		//XXX: ^^ breaks hpout restore
 		snd_soc_component_update_bits(component, RT5640_PWR_ANLG1,
 			RT5640_PWR_LM, RT5640_PWR_LM);
 		snd_soc_component_update_bits(component, RT5640_OUTPUT,
-- 
2.35.3

